### ISSUE TITLE
Support for comparing the difference between two images

### ISSUE THREAD

**Issue by danbao** [THUDM/CogVLM#308]
### Feature request  / 功能建议

**Objective**: Introduce a feature in CogVLM that allows users to upload two images and receive a detailed comparison of their differences.
**Functionality**: The tool should analyze both images in terms of color, composition, subject matter, and other visual elements. It should then provide a comprehensive report highlighting similarities and differences.
**User Input**: Users should be able to upload two images directly or provide URLs to the images.
**Output Format**: A side-by-side comparison report, possibly including marked areas of difference, textual descriptions of dissimilarities, and a summary of key comparison points.
**Usability**: Easy-to-use interface and API.

### Motivation / 动机

**Enhanced User Experience**: This feature would greatly benefit users needing to analyze visual changes, discrepancies, or progress over time between two images.
**Accessibility**: Such a feature would make advanced image analysis more accessible to a broader audience, reducing the reliance on specialized or technical software.
**Innovative Edge**: Incorporating this feature would position CogVLM as a more versatile and cutting-edge tool in the field of image processing and analysis.

### Your contribution / 您的贡献

N/A

**Comment by aldoz-mila** [comment]
> This is more difficult. In the open source model, the training set does not jointly process multiple images, so it is not possible to use the current model to match multiple images.

Do you have an intuition of how much notion of localization the model has for the image? I was wondering if merging 2 images into one on the horizontal side and referring to the "left" and "right" sides in the prompting could be a hacky way around this current limitation? I would be curious to explore this.

**Comment by zRzRzRzRzRzRzR** [comment]
You are right. Splicing two pictures and then judging by a large model is possible with our existing model. Our cogagent model supports high-resolution pictures, so you can splice the two pictures and The model is required to analyze and guess this stitched picture, which will satisfy some basic tasks. Just like the picture I provided

<img width="963" alt="image" src="https://github.com/THUDM/CogVLM/assets/93239683/6aa13a76-8294-4626-a675-4c20534ac62e">

**Comment by danbao** [comment]
> > This is more difficult. In the open source model, the training set does not jointly process multiple images, so it is not possible to use the current model to match multiple images.
> 
> Do you have an intuition of how much notion of localization the model has for the image? I was wondering if merging 2 images into one on the horizontal side and referring to the "left" and "right" sides in the prompting could be a hacky way around this current limitation? I would be curious to explore this.

This is indeed a great idea, but the experimental results are not very ideal. Especially for the comparison of product pages (maybe due to the fact of the relevant training set)

**Comment by zRzRzRzRzRzRzR** [comment]
Oh, that's for sure because our open-source model lacks training on these datasets, and we're still trying to provide a feasible fine-tuning solution. Also, you can try our new [API](https://open.bigmodel.cn) after our press conference tomorrow
